cmd/nerdctl/load_encrypt_linux_test.go

```diff
@@ -0,0 +1,70 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"testing"
+
+	"github.com/containerd/nerdctl/pkg/rootlessutil"
+	"github.com/containerd/nerdctl/pkg/testutil"
+	"github.com/containerd/nerdctl/pkg/testutil/testregistry"
+	"gotest.tools/v3/assert"
+)
+
+func TestLoadEncryptImage(t *testing.T) {
+	testutil.DockerIncompatible(t)
+	keyPair := newJWEKeyPair(t)
+	defer keyPair.cleanup()
+
+	// copy temp private key to default directory, from default config.toml --decryption-keys-path value
+	ocicryptDefaultPath := "/etc/containerd/ocicrypt/keys"
+	if rootlessutil.IsRootless() {
+		ocicryptDefaultPath = "~/.config/containerd/ocicrypt/keys"
+	}
+	err := os.MkdirAll(ocicryptDefaultPath, 0755)
+	assert.NilError(t, err)
+	exec.Command("cp", "-f", keyPair.prv, ocicryptDefaultPath)
+	prvName := filepath.Base(keyPair.prv)
+	etcPrvPath := filepath.Join(ocicryptDefaultPath, prvName)
+	defer os.Remove(etcPrvPath)
+
+	base := testutil.NewBase(t)
+	tID := testutil.Identifier(t)
+	reg := testregistry.NewPlainHTTP(base, 5000)
+	defer reg.Cleanup()
+	base.Cmd("pull", testutil.CommonImage).AssertOK()
+	encryptImageRef := fmt.Sprintf("127.0.0.1:%d/%s:encrypted", reg.ListenPort, tID)
+	tmpPath := path.Dir(keyPair.pub)
+	encryptImageTar := path.Join(tmpPath, fmt.Sprintf("%s.encrypted.tar", tID))
+	defer os.Remove(encryptImageTar)
+	defer base.Cmd("rmi", encryptImageRef).Run()
+	base.Cmd("image", "encrypt", "--recipient=jwe:"+keyPair.pub, testutil.CommonImage, encryptImageRef).AssertOK()
+	base.Cmd("save", "--output", encryptImageTar, encryptImageRef).AssertOK()
+
+	base.Cmd("rmi", encryptImageRef).Run()
+
+	base.Cmd("load", "--input", encryptImageTar).AssertOK()
```

Comment:
Can we add a load failure case?

Reply:
:+1:. 

Even if an image is not decryptable on `load`, the encrypted image should be still loaded.
User may attempt to use `nerdctl image decrypt` to decrypt the image later.

Reply:
(Also, `nerdctl load` should have an option to skip decryption )

Reply:
> Can we add a load failure case?

done.

> Also, nerdctl load should have an option to skip decryption

add `--decrypt` option for unpack encrypt image. default is `false`

Reply:
Please update the README.md.